scripts/sanity_chk/sanitylib.py

```diff
@@ -2845,132 +2845,151 @@ def process_log(log_file):
-                                eleTestcase,
-                                'error',
-                                type="failure",
-                                message="failed")
-                        p = os.path.join(self.outdir, instance.platform.name, instance.testcase.name)
-                        log_file = os.path.join(p, "handler.log")
-                        el.text = self.process_log(log_file)
-
-                    elif instance.results[k] == 'SKIP':
-                        el = ET.SubElement(
-                            eleTestcase,
-                            'skipped',
-                            type="skipped",
-                            message="Skipped")
-            else:
-                eleTestcase = ET.SubElement(eleTestsuite, 'testcase',
-                    classname="%s:%s" % (instance.platform.name, instance.testcase.name),
-                    name="%s" % (instance.testcase.name),
-                    time="%f" % handler_time)
-                if instance.status in ["failed", "timeout"]:
-                    failure = ET.SubElement(
-                        eleTestcase,
-                        'failure',
-                        type="failure",
-                        message=instance.reason)
-                    p = ("%s/%s/%s" % (self.outdir, instance.platform.name, instance.testcase.name))
-                    bl = os.path.join(p, "build.log")
-                    hl = os.path.join(p, "handler.log")
-                    log_file = bl
-                    if instance.reason != 'Build error':
-                        if os.path.exists(hl):
-                            log_file = hl
+                            fails += 1
+                else:
+                    if instance.status in ["failed", "timeout"]:
+                        if instance.reason in ['build_error', 'handler_crash']:
+                            errors += 1
                         else:
-                            log_file = bl
+                            fails += 1
+                    elif instance.status == 'skipped':
+                        skips += 1
+                    else:
+                        passes += 1
 
-                    failure.text = self.process_log(log_file)
+            total = (errors + passes + fails + skips)
+            # do not produce a report if no tests were actually run (only built)
+            if total == 0:
+                continue
 
-                elif instance.status == "skipped":
-                    ET.SubElement(eleTestcase, 'skipped', type="skipped", message="Skipped")
+            run = p
+            eleTestsuite = None
 
+            # When we re-run the tests, we re-use the results and update only with
+            # the newly run tests.
+            if os.path.exists(filename) and append:
+                eleTestsuite = eleTestsuites.findall(f'testsuite/[@name="{p}"]')[0]
+                eleTestsuite.attrib['failures'] = "%d" % fails
```

Comment:
I think it would improve readability it strings would formatted using f-strings as in !22152
So for this line:
```
eleTestsuite.attrib['failures'] = f"{fails}"
```

Reply:
That seems like a more global cleanup to these scripts that should be looked at.